tsdb/wlog/wlog_test.go

```diff
@@ -190,9 +220,10 @@ func TestWALRepair_ReadingError(t *testing.T) {
 			}
 			require.NoError(t, r.Err())
 			require.Len(t, result, test.intactRecs, "Wrong number of intact records")
+			require.Len(t, test.intactRecIndex, len(result), "Wrong number of intact records")
 
 			for i, r := range result {
-				if !bytes.Equal(records[i], r) {
+				if !bytes.Equal(records[test.intactRecIndex[i]], r) {
```

Comment:
Can you explain why you change this assertion?

Reply:
Actually, having read the PR again, I think this change should not be made, since in this test the expected sequence is always contiguous between 0 inclusive and `test.intactRecs` exclusive.